file-upload/file-upload-java-sbt/README.md

```diff
@@ -0,0 +1,91 @@
+
+```bash
+sbt runAll
+```
+
+On a separate terminal, use `curl` to POST a file (in this example we're posting `build.sbt`:
+
+```bash
+curl -X POST -F "data=@build.sbt" -v  http://localhost:9000/api/files
+```
+
+You can also exercise a regular Lagom endpoint that coexists with the file upload controller:
+
+```bash
+curl -X POST -H "Content-Type: text/plain" -d  "hello world" http://localhost:9000/api/echo
+```
+
+
+
+## Code details: TOC
+
+The changes required on a Lagom service to handle File upload are:
+
+1. Add a new [Play controller](https://www.playframework.com/documentation/2.6.x/JavaActions) named `FileUploadController`
+2. Create a new [`routes` file](https://www.playframework.com/documentation/2.6.x/JavaRouting) and add a new route pointing to our side-car `FileUploadController`
+3. Fall back to Play's routing instead of using Lagom's default `Router`
+
+There are few more details on the recipe worth mentioning:
+
+1. Service ACLs setup
+2. Automated Tests
+
+
+##### Code details: Add a new Play controller
+
+This steps only requires creating a new controller based on [Play's Upload File example](https://www.playframework.com/documentation/2.6.x/JavaFileUpload). The final controller is [`FileUploadController`](./fileupload-impl/src/main/java/com/example/play/controllers/FileUploadController.java)
+
+##### Code details: Create a new `routes` file
+  * we then create a file named `routes` in `./fileupload-impl/src/main/resources`. You can learn more about that file and it's syntax on the [docs](https://www.playframework.com/documentation/2.6.x/JavaRouting). In our case we want the `routes` file to contain only one entry:
+
+            POST    /api/files      com.example.play.controllers.FileUploadController.upload()
+
+##### Code details: Fall back to Play's routing instead of using Lagom's default `Router`
+
+  * in the `build.sbt` project definiton we have to make a few changes. All changes need to be applied on the project definition of the implementation module. First we enable the [`PlayJava`](./build.sbt#L21) plugin that will let us use `routes`-based routing. Then, we're configuring Play to use the [`Injected routes generator`](https://www.playframework.com/documentation/2.6.x/JavaDependencyInjection#Injected-routes-generator). Finally, we  need to [disable](./build.sbt#L22) the `PlayLayoutPlugin` because we're using `sbt`'s default project structure instead of [Play's project structure](https://www.playframework.com/documentation/2.6.x/Anatomy). The last step is only required because we want to maintain Lagom's project structure.
+  * next, in [`FileUploadModule`](./fileupload-impl/src/main/java/com/example/FileUploadModule.java#L15) we add a additional router that the `PlayJava` sbt plugin will create for us from the `src/main/resources/routes` file we created above. This class `Routes` is created on the fly by the `PlayJava` plugin considering the `InjectedRoutesGenerator` setting in `build.sbt`.
```

Comment:
It'd be good to specify that `additionalRouter` is API added in Lagom 1.5

Reply:
Don't come up with anything better 😄 
> next, in `FileUploadModule` we add a additional router (_Lagom 1.5+_) ...

Reply:
My concern is that this recipe is the Java counterpart of a Scala recipe and the most important detail (composing routers) differs dramatically because this recipe uses an feature introduced in Lagom 1.5. I'd rather make this very obvious in:

 * the line of `README` here ( line `69`)
 * the intro in `README` -> state this recipe differs from the scala one (just next to stating this recipe requires 1.5+)
 * the code in the `Module` indicating `additionalRouter` is API introduced in Lagom 1.5 so that users landing on that code directly (stale link, googling, etc...) can quickly spot it's a 1.5+ feature.